.github/workflows/internal_version_check.yaml

```
@@ -0,0 +1,33 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+name: Internal Version Check
+
+concurrency:
+  group: "${{ github.workflow }}-${{ github.ref }}"
+  cancel-in-progress: true
+
+on:
+  pull_request:
+
+jobs:
+  check-incremented-version:
+    name: Check Internal Version Incremented
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - name: Compare with internal version on main branch
+        run: |
+          NEW_VERSION="$(< charm_internal_version)"
+          mkdir repo; cd repo
+          git clone https://github.com/canonical/mongodb-operator.git
+          cd mongodb-operator
+          OLD_VERSION="$(< charm_internal_version)"
```

This will fail if the file doesn't exist already, we should perhaps set a default to `0` if it's not there.
```suggestion
          OLD_VERSION="$([ -f charm_internal_version ] && cat charm_internal_version || echo 0)"
```